src/librbd/io/ObjectRequest.cc

```diff
@@ -900,6 +900,11 @@ void ObjectListSnapsRequest<I>::handle_list_snaps(int r) {
     if (end_snap_id <= first_snap_id) {
       // don't include deltas from the starting snapshots, but we iterate over
       // it to track its existence and size
+      // Note: if the image is a clone and a copyup was performed on the
+      // object, the older snaps will be updated with the parent data.
+      // When called by snapshot based rbd-mirror, the diff in the older snap is ignored
+      // as it was processed earlier, causing the snapshot_delta to not include
+      // the parent data.
```

Comment:
I am not sure we need this comment here.

Reply:
I think it helps explain what happens  - it took me a long time to figure this out.